extensions/amp-animation/0.1/web-animations.js

```diff
@@ -126,7 +127,7 @@ export class WebAnimationRunner {
       // Apply vars.
       if (request.vars) {
         for (const k in request.vars) {
-          request.target.style.setProperty(k, String(request.vars[k]));
+          setImportantStyles(request.target, {[k]: String(request.vars[k])});
```

Comment:
@aghassemi FYI

Reply:
Right, that is what I did initially but Will was stating that we use !important. So in a discussion with @Gregable and @choumx about supporting inline styles in amp components, I was in favor of not supporting it..as I was concerned about the scenario where the user decides to style the component which then gets clobbered by internal style mutations but that wouldn't stop them from making style changes to the component via css inheritance..if the amp-img was wrapped in a div that was styled for example..unless we decide to sanitize that? so if we allowed inline styles in amp components then the !important will play a significant role in ensuring that the web animation works and behaves accordingly.

Reply:
This is actually orthogonal to the inline styles feature. The issue (#14178) is just that sometimes publishers can unintentionally override AMP styles if we don't set them with `!important` in the runtime.

Reply:
> The issue (#14178) is just that sometimes publishers can unintentionally override AMP styles if we don't set them with !important in the runtime.

It's not publishers. The only time `!important` is necessary is when _we_ set an `!important` property in our provided CSS files. Otherwise, inline style always override inheritance and have infinite specificity.

Reply:
Right, thanks.